src/utils/prim-lexing.fs

```diff
@@ -156,14 +156,21 @@ namespace Internal.Utilities.Text.Lexing
             new LexBuffer<'Char>(filler)
               
         // A full type signature is required on this method because it is used at more specific types within its own scope
-        static member FromArray (s: 'Char[]) : LexBuffer<'Char> = 
+        // Important: This method takes ownership of the array
+        static member FromArrayNoCopy (buffer: 'Char[]) : LexBuffer<'Char> = 
             let lexBuffer = new LexBuffer<'Char>(fun _ -> ())
-            let buffer = Array.copy s 
```

Comment:
please remove the ; while you are touching this

Reply:
Done, were left due to oversight.